source/frontend/src/features/mapSideBar/compensation/detail/CompensationRequisitionHistoricalDetailContainer.tsx

```diff
@@ -0,0 +1,350 @@
+import { ApiGen_Concepts_FinancialCode } from '@/models/api/generated/ApiGen_Concepts_FinancialCode';
+import { ApiGen_Concepts_Lease } from '@/models/api/generated/ApiGen_Concepts_Lease';
+import { ApiGen_Concepts_Product } from '@/models/api/generated/ApiGen_Concepts_Product';
+import { ApiGen_Concepts_Project } from '@/models/api/generated/ApiGen_Concepts_Project';
+import { isValidId } from '@/utils';
+
+import { useGenerateH120 } from '../../acquisition/common/GenerateForm/hooks/useGenerateH120';
+import { CompensationRequisitionDetailViewProps } from './CompensationRequisitionDetailView';
+
+export interface ICompensationRequisitionHistoricalDetailContainerProps {
+  fileType: ApiGen_CodeTypes_FileTypes;
+  parentFileId: number;
+  compensationRequisitionId: number;
+  time: string;
+  clientConstant: string;
+  loading: boolean;
+  setEditMode: (editMode: boolean) => void;
+  View: React.FunctionComponent<CompensationRequisitionDetailViewProps>;
+}
+
+export const CompensationRequisitionHistoricalDetailContainer: React.FunctionComponent<
+  React.PropsWithChildren<ICompensationRequisitionHistoricalDetailContainerProps>
+> = ({
+  fileType,
+  parentFileId,
+  compensationRequisitionId,
+  time: almostTime,
+  clientConstant,
+  loading,
+  setEditMode,
+  View,
+}) => {
+  const onGenerate = useGenerateH120();
+
+  const adjustedTime = useMemo(() => {
+    const momentTime = moment(almostTime);
+    return momentTime.add(1, 'seconds').toISOString();
+  }, [almostTime]);
+
+  const [parentFile, setParentFile] = useState<
+    ApiGen_Concepts_Lease | ApiGen_Concepts_AcquisitionFile | null
+  >(null);
+  const [project, setProject] = useState<ApiGen_Concepts_Project | null>(null);
+  const [alternateProject, setAlternateProject] = useState<ApiGen_Concepts_Project | null>(null);
+  const [product, setProduct] = useState<ApiGen_Concepts_Product | null>(null);
+
+  const [timedCompensation, setTimedCompensation] =
+    useState<ApiGen_Concepts_CompensationRequisition | null>(null);
+
+  const [financialActivityCodes, setFinancialActivityCodes] = useState<
+    ApiGen_Concepts_FinancialCode[]
+  >([]);
+  const [chartOfAccountCodes, setChartOfAccountCodes] = useState<ApiGen_Concepts_FinancialCode[]>(
+    [],
+  );
+  const [yearlyFinancialCodes, setYearlyFinancialCodes] = useState<ApiGen_Concepts_FinancialCode[]>(
+    [],
+  );
+  const [responsibilityCentreCodes, setResponsibilityCentreCodes] = useState<
+    ApiGen_Concepts_FinancialCode[]
+  >([]);
+
+  const [compensationRequisitionProperties, setCompensationRequisitionProperties] = useState<
+    ApiGen_Concepts_FileProperty[]
+  >([]);
+
+  const [acquisitionPayees, setAcquisitionPayees] = useState<ApiGen_Concepts_CompReqAcqPayee[]>([]);
+  const [leasePayees, setLeasePayees] = useState<ApiGen_Concepts_CompReqLeasePayee[]>([]);
+
+  const {
+    getFinancialActivityCodeTypes: {
+      execute: fetchFinancialActivities,
```

Comment:
nit: all of these wrapped repository functions provide a data field in the response, which would save you from having to explicitly create all of these state vars and set them - might be more concise.